File: Buddy/buddy-flutter/lib/models/pet_need.dart
import 'pet.dart';

enum PetNeed {
  hungry,
  thirsty,
  sleepy,
  dirty,
  bored;

  String get emoji => switch (this) {
        PetNeed.hungry  => '🍖',
        PetNeed.thirsty => '💧',
        PetNeed.sleepy  => '💤',
        PetNeed.dirty   => '🧼',
        PetNeed.bored   => '🎾',
      };

  String get label => switch (this) {
        PetNeed.hungry  => 'Hambre',
        PetNeed.thirsty => 'Sed',
        PetNeed.sleepy  => 'Sueño',
        PetNeed.dirty   => 'Sucio',
        PetNeed.bored   => 'Aburrido',
      };

  String get actionLabel => switch (this) {
        PetNeed.hungry  => 'Alimentar',
        PetNeed.thirsty => 'Dar agua',
        PetNeed.sleepy  => 'Dormir',
        PetNeed.dirty   => 'Bañar',
        PetNeed.bored   => 'Jugar',
      };

  int severity(Pet pet) => switch (this) {
        PetNeed.hungry  => 100 - pet.stats.hunger,
        PetNeed.thirsty => 100 - pet.stats.thirst,
        PetNeed.sleepy  => 100 - pet.stats.energy,
        PetNeed.dirty   => 100 - pet.stats.hygiene,
        PetNeed.bored   => 100 - pet.stats.happiness,
      };
}

File: Buddy/buddy-ios/Sources/buddy/Models/PetNeed.swift
import Foundation

enum PetNeed: String, Identifiable, Hashable, CaseIterable {
    case hungry, thirsty, sleepy, dirty, bored

    var id: String { rawValue }

    var emoji: String {
        switch self {
        case .hungry:  "🍖"
        case .thirsty: "💧"
        case .sleepy:  "💤"
        case .dirty:   "🧼"
        case .bored:   "🎾"
        }
    }

    var label: String {
        switch self {
        case .hungry:  "Hambre"
        case .thirsty: "Sed"
        case .sleepy:  "Sueño"
        case .dirty:   "Sucio"
        case .bored:   "Aburrido"
        }
    }

    /// Verb for the contextual action button.
    var actionLabel: String {
        switch self {
        case .hungry:  "Alimentar"
        case .thirsty: "Dar agua"
        case .sleepy:  "Dormir"
        case .dirty:   "Bañar"
        case .bored:   "Jugar"
        }
    }

    func severity(for pet: Pet) -> Int {
        switch self {
        case .hungry:  100 - pet.stats.hunger
        case .thirsty: 100 - pet.stats.thirst
        case .sleepy:  100 - pet.stats.energy
        case .dirty:   100 - pet.stats.hygiene
        case .bored:   100 - pet.stats.happiness
        }
    }
}
